pkg/controller/nodeipam/ipam/range_allocator_test.go

```diff
@@ -212,24 +221,27 @@ func TestAllocateOrOccupyCIDRSuccess(t *testing.T) {
 				},
 				Clientset: fake.NewSimpleClientset(),
 			},
-			clusterCIDRs: func() []*net.IPNet {
-				_, clusterCIDRv4, _ := net.ParseCIDR("127.123.234.0/8")
-				_, clusterCIDRv6, _ := net.ParseCIDR("ace:cab:deca::/8")
-				_, clusterCIDRv4_2, _ := net.ParseCIDR("10.0.0.0/8")
-				return []*net.IPNet{clusterCIDRv4, clusterCIDRv6, clusterCIDRv4_2}
-			}(),
-			serviceCIDR: func() *net.IPNet {
-				_, serviceCIDR, _ := net.ParseCIDR("127.123.234.0/26")
-				return serviceCIDR
-			}(),
-			secondaryServiceCIDR: nil,
+			allocatorParams: CIDRAllocatorParams{
+				ClusterCIDRs: func() []*net.IPNet {
+					_, clusterCIDRv4, _ := net.ParseCIDR("127.123.234.0/8")
+					_, clusterCIDRv6, _ := net.ParseCIDR("ace:cab:deca::/84")
+					_, clusterCIDRv4_2, _ := net.ParseCIDR("10.0.0.0/8")
+					return []*net.IPNet{clusterCIDRv4, clusterCIDRv6, clusterCIDRv4_2}
+				}(),
+				ServiceCIDR: func() *net.IPNet {
+					_, serviceCIDR, _ := net.ParseCIDR("127.123.234.0/26")
+					return serviceCIDR
+				}(),
+				SecondaryServiceCIDR: nil,
+				NodeCIDRMaskSizes:    []int{24, 98, 24},
```

Comment:
Should we add a test where the node mask CIDR order is different from the cluster CIDR order? Does nodeipam  controller even check that here or is that something that is only validated at kube-controller-manager?

Reply:
Right, that is validated in `kube-controller-manager` as that is the one that generates the node cidr mask sizes list based on the cluster cidr list.